src/libstd/net/ip.rs

```diff
@@ -1488,20 +1496,25 @@ impl Ipv6Addr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Display for Ipv6Addr {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Long enough for the longest possible Ipv6: 45 bytes for an IPv4 compatible address
+        const IPV6_BUF_LEN: usize = 48;
```

Comment:
45 <-> 48?

Reply:
So it turns out that the first number, 45, comes from IPv6 mapped IPv4 addresses, but we only generate in that format if the first few segments are 0, so the real max length is only 39